grpc.go

```diff
@@ -81,9 +81,59 @@ func handleError(wait bool, err error) error {
 	return err
 }
 
-// Online resources, nbResources specifies the maximum number of resources to online.
-// If nbResources is <= 0 then there is no limit and all resources are connected.
-// Returns the number of resources connected.
+// Offline resources, nbResources specifies the maximum number of resources to offline.
+// If nbResources is <= 0 then there is no limit and all resources are disconnected.
+// Returns the number of resources disconnected.
+func offlineResources(resource onlineResource, nbResources int32) (uint32, error) {
+	// Get the bitmap of the online cpus
+	onlineCPUBitmapPath := filepath.Join(resource.sysfsOnlinePath, "online")
+	status, err := ioutil.ReadFile(onlineCPUBitmapPath)
+	if err != nil {
+		return 0, err
+	}
+
+	// Get the last online cpu, e.g. 0,3,4-7  cpu7 is the last online cpu
```

Comment:
I still find this block confusing - if you just need the last cpu -- assuming the `online` file is guaranteed to be ordered -- why do you need to do the split?

I think it would help if you split `offlineResources()` into a number of functions like:

- `getCPUsToOffline(...) []int`
- `offlineCPU(cpuNum int32) error`
- ...

Then, you can add a new unit tests to assert which cpus should be offlined and which should be left online. A table-based approach like the following would be ideal for testing `getCPUsToOffline()` for example:

- https://github.com/kata-containers/agent/blob/master/config_test.go#L106..L137

Reply:
hi @jodh-intel Thanks for the comments, but seems my proposal hasn't been accepted yet :(